src/core/qgsvectorlayerexporter.cpp

```diff
@@ -120,6 +121,22 @@ QgsVectorLayerExporter::QgsVectorLayerExporter( const QString &uri,
     return;
   }
 
+  // If the result is a geopackage layer and there is already a field name FID requested which
+  // might contain duplicates, make sure to generate a new field with a unique name instead
+  // that will be filled by ogr with unique values.
+
+  // HACK sorry
+  const QString path = QgsProviderRegistry::instance()->decodeUri( QStringLiteral( "ogr" ), uri ).value( QStringLiteral( "path" ) ).toString();
```

Comment:
Could we get a specific unit test covering this?

Reply:
"this"?

Reply:
done